src/main/java/org/truffleruby/core/hash/SetNode.java

```diff
@@ -22,6 +22,9 @@
 
 @ImportStatic(HashGuards.class)
 public abstract class SetNode extends RubyContextNode {
+    /* On execution SetNode performs Hash#[]= and returns true if the key is newly added to hash, and returns false if
+     * the key was found in the hash prior. This is different from the return value of Hash#[]=, which returns the right
+     * hand side value, however this change does a set and returns the result of get without an extra lookup. */
```

Comment:
This should also be a JavaDoc comment above the class declaration like for DeleteLastNode.